src/dll/Hooks/ReportErrorCode.cpp

```diff
@@ -0,0 +1,73 @@
+#include "ReportErrorCode.hpp"
+#include "Addresses.hpp"
+#include "App.hpp"
+#include "Hook.hpp"
+#include "stdafx.hpp"
+
+namespace
+{
+bool isAttached = false;
+
+void _State_SessionActive_ReportErrorCode(uintptr_t);
+Hook<decltype(&_State_SessionActive_ReportErrorCode)> State_SessionActive_ReportErrorCode(
+    0x7FA31576, &_State_SessionActive_ReportErrorCode);
+
+void _State_SessionActive_ReportErrorCode(uintptr_t aThis)
+{
+    auto errorCode = *reinterpret_cast<uint32_t*>(aThis + 0x88u);
+
+    if (errorCode > 0u)
+    {
+        static const auto type = RED4ext::CRTTISystem::Get()->GetEnum("gsmStateError");
+
+        spdlog::error("=======");
+        spdlog::error("Session error, error code: {}", type->hashList[errorCode].ToString());
```

Comment:
I assume `gsm` is GameStateManager? If so I would suggest a message like:

```suggestion
        spdlog::error("A game session error occurred. Error: {} ({})", type->hashList[errorCode].ToString(), errorCode);
```

Reply:
Likely GameStateMachine, as psiberx said